Language/Classes/Superinterfaces/same_generic_class_t01.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if a class C has two superinterfaces
+/// that are different instantiations of the same generic class
+///
+/// @description Checks that it is a compile-time error if a class `C` has two
+/// superinterfaces that are different instantiations of the same generic class.
+/// @author sgrekhov22@gmail.com
+
+class A<T> {}
+
+class C1 implements A<int>, A {}
```

Comment:
This is an interesting case! I'm afraid we might get the wrong error here.

The standard example would rely on indirect superinterfaces (so `C1 implements A<int>, OtherClass` and `OtherClass implements A<String>`).

Just trying an example: `class B extends A, A {}`. That one does actually give rise to an error, "'A' can only be implemented once", even though there is no discrepancy with respect to the actual type arguments.

So we might have a case where that other error is reported, but (due to a bug) the conflict between `A<int>` and `A<dynamic>` is not detected and not reported, but the test apparently succeeds.

Found the error in the spec: https://github.com/dart-lang/language/blob/cc2c933b1ae5a2c837f89ca4490040ff158a676f/specification/dartLangSpec.tex#LL4728C5-L4728C5.

That error shouldn't apply in this case (they are not the same type), so it's actually very good that we have this case as well.

But the other cases should test the "standard" situation (the one that will actually occur in code) where the two different generic instantiations of `A` occur in two different superinterfaces, or one occurs on the class itself, and the other one occurs on the declaration of a superinterface, direct or indirect.

Reply:
Also, the error should be detected when `A<int>` occurs on a superclass and `A<String>` occurs on a mixin in the `with` clause, and the other combinations (extends/with, extends/implements, with/implements, extends/extends, with/with, implements/implements).

Well, perhaps we can't do extends/extends, because this implies that we're inheriting from a class C1 and from a class C2 (both superclasses), and they both implement `A<Something>`, but it's "something different", and this means that the most specific one (`C2` if `C2 <: C1`) is already an error. And we don't want to test derived errors.

But I believe the other combinations can be expressed without having a derived error.

Reply:
Thanks!